### ISSUE TITLE
Network unreachable errors

### ISSUE THREAD

**Issue by muellermartin** [cdown/tzupdate#17]
If I run `tzupdate -p` following errors are shown (but the timzone gets detected):

```
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 53, in get_timezone_for_ip
    api_response = requests.get(api_url).json()
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='ip-api.com', port=80): Max retries exceeded with url: /json/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f8685f06a90>: Failed to establish a new connection: [Errno 111] Connection refused',))
Process Process-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 53, in get_timezone_for_ip
    api_response = requests.get(api_url).json()
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPSConnectionPool(host='freegeoip.net', port=443): Max retries exceeded with url: /json/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f8685f06ad0>: Failed to establish a new connection: [Errno 101] Network unreachable',))
Detected timezone is Europe/Berlin.
```

Additional information:

- OS: Alpine Linux 3.5.0
- Python: Python 2.7.13

**Comment by cdown** [comment]
What's the output with `--debug`? It looks like you have some network issues unrelated to tzupdate.

**Comment by muellermartin** [comment]
This is the full output of `tzupdate -p --debug` as requested:

```
DEBUG:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): ip-api.com
DEBUG:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): geoip.nekudo.com
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
DEBUG:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): freegeoip.net
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 53, in get_timezone_for_ip
    api_response = requests.get(api_url).json()
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
Process Process-2:
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 53, in get_timezone_for_ip
    api_response = requests.get(api_url).json()
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    raise ConnectionError(e, request=request)
    resp = self.send(prep, **send_kwargs)
ConnectionError: HTTPConnectionPool(host='ip-api.com', port=80): Max retries exceeded with url: /json/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7faa3a85ba90>: Failed to establish a new connection: [Errno 111] Connection refused',))
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPSConnectionPool(host='freegeoip.net', port=443): Max retries exceeded with url: /json/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7faa3a85bad0>: Failed to establish a new connection: [Errno 101] Network unreachable',))
DEBUG:requests.packages.urllib3.connectionpool:http://geoip.nekudo.com:80 "GET /api/ HTTP/1.1" 200 None
DEBUG:tzupdate:API response from http://geoip.nekudo.com/api/: {u'ip': u'93.215.45.8', u'city': False, u'location': {u'latitude': 51.2993, u'accuracy_radius': 200, u'longitude': 9.491}, u'country': {u'code': u'DE', u'name': u'Germany'}}
Process Process-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 62, in get_timezone_for_ip
    api_response.get(service.error_key, 'Unspecified API error.')
TimezoneAcquisitionError: Unspecified API error.
Traceback (most recent call last):
  File "/usr/bin/tzupdate", line 11, in <module>
    load_entry_point('tzupdate==1.1.0', 'console_scripts', 'tzupdate')()
  File "/usr/lib/python2.7/site-packages/tzupdate.py", line 216, in main
    "No response from any API in {} seconds".format(args.timeout)
tzupdate.TimezoneAcquisitionError: No response from any API in 5.0 seconds
```

In this run no thread seems to be successful determing the timezone. Network seems to work basically (tried pinging some DNS hosts and tested HTTP briefly with curl).

Additionaly infomation: tzupdate runs in a Docker container on macOS in the native hypervisor (Hypervisor Framework).

**Comment by cdown** [comment]
```
Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7faa3a85bad0>: Failed to establish a new connection: [Errno 101] Network unreachable',)
```

"Network unreachable" means you have some routing problem (judging by the sporadic nature of the failures, perhaps you have a corrupt routing table). We don't do anything specific to the network in tzupdate -- I can't think of any way that this could be specific to tzupdate.

If you believe otherwise, please attach a pcap of the affected interface :-)

**Comment by muellermartin** [comment]
After you pointed out a network issue I was skeptical first (especially after testing a few hosts), but then I realized that I'm using a DNS blocklist ([StevenBlack/hosts](https://github.com/StevenBlack/hosts)). A quick grep confirmed that at least some of the Geo IP domains are on this list and disabling it fixed the issue (see also StevenBlack/hosts#210). Thanks for the hint! I would consider this as resolved.
